### PR TITLE
Added Spring4Shell.bcheck rule

### PR DESCRIPTION
**Pull request by GiriRaj249**
## About

Added a rule to check for Spring4Shell vulnerability by targeting GET requests. Following are the conditions that I have used to validate the existence of the bug :

- status code is "400"
- response body should contain `java.lang.IllegalArgumentException` error

### CODE REVIEW

File: vulnerability-classes/injection/Spring4Shell.bcheck [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+define:
+    spring4shell_payload = `class.module.classLoader.URLs%5B0%5D=0`
+    issueDetail = `The server has returned a response status code as "400" and along with "java.lang.IllegalArgumentException" error on the response body denoting that there is possibility for Spring4shell vulnerability`
+    issueRemediation = "Make sure you are up to date with patches and follow the remediation steps for CVE-2022-22965."
+
+given request then
+    send request called check :
+        appending queries: `{spring4shell_payload}`
+
+		if {check.response.status_code} is "400" and "java.lang.IllegalArgumentException" in {check.response.body} then
```
Comment (olliewuk) [inline vulnerability-classes/injection/Spring4Shell.bcheck:17]: can you fix in identing please?

**Review by olliewuk — CHANGES_REQUESTED** [review]

File: vulnerability-classes/injection/Spring4Shell.bcheck (hunk 1 above)
Comment (GiriRaj249) [reply vulnerability-classes/injection/Spring4Shell.bcheck]: Yes, I have fixed the indentation on the latest commit.

Comment (GiriRaj249) [reply vulnerability-classes/injection/Spring4Shell.bcheck]: Hey, olliewuk!! good day. Any update on the commit? I have requested a review

**Review by olliewuk — APPROVED** [review]

**Comment by GiriRaj249** [comment]
Hey Team,

I got 1 approval from Olliewuk, but this merge required at least 2 approvals. When can I expect my commit to be pushed on the BCheck repo ?

Awaiting your approval/response. Thanks in advance

**Comment by GiriRaj249** [comment]
Hey Team Portswigger,

Any update that i need to be aware off?

**Review by Hannah-PortSwigger — APPROVED** [review]

**Pull request merged**